Exception generated by card limiter that prevent cards to be sent

Exception that prevent to send card with a specific account 

```
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Long.longValue()" because the return value of "java.util.List.get(int)" is null
        at org.opfab.cards.publication.ratelimiter.CardSendingLimiter.isLimitReached(CardSendingLimiter.java:55) ~[!/:4.3.2.RELEASE]
        at org.opfab.cards.publication.ratelimiter.CardSendingLimiter.isNewSendingAllowed(CardSendingLimiter.java:37) ~[!/:4.3.2.RELEASE]
        at org.opfab.cards.publication.services.CardProcessingService.processCard(CardProcessingService.java:104) ~[!/:4.3.2.RELEASE]
        at org.opfab.cards.publication.controllers.CardController.createCard(CardController.java:71) ~[!/:4.3.2.RELEASE]
        at jdk.internal.reflect.GeneratedMethodAccessor109.invoke(Unknown Source) ~[na:na]
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
        at java.base/java.lang.reflect.Method.invoke(Method.java:568) ~[na:na]
        at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:255) ~[spring-web-6.1.6.jar!/:6.1.6]
        at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:188) ~[spring-web-6.1.6.jar!/:6.1.6]
        at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:118) ~[spring-webmvc-6.1.6.jar!/:6.1.6]
        at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:926) ~[spring-webmvc-6.1.6.jar!/:6.1.6]
        at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:831) ~[spring-webmvc-6.1.6.jar!/:6.1.6]
        at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) ~[spring-webmvc-6.1.6.jar!/:6.1.6]
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1089) ~[spring-webmvc-6.1.6.jar!/:6.1.6]
        at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:979) ~[spring-webmvc-6.1.6.jar!/:6.1.6]
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1014) ~[spring-webmvc-6.1.6.jar!/:6.1.6]
```

It seems to be a concurrency issue

--- a/services/cards-publication/src/main/java/org/opfab/cards/publication/ratelimiter/CardSendingLimiter.java
+++ b/services/cards-publication/src/main/java/org/opfab/cards/publication/ratelimiter/CardSendingLimiter.java
@@ -1,4 +1,4 @@
-/* Copyright (c) 2023, RTE (http://www.rte-france.com)
+/* Copyright (c) 2023-2024, RTE (http://www.rte-france.com)
  * See AUTHORS.txt
  * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
@@ -33,13 +33,17 @@ public class CardSendingLimiter {
 
     public boolean isNewSendingAllowed(String login) {
         List<Long> cardSendings = getCardSendings(login.toUpperCase());
-
-        if (!isLimitReached(cardSendings)) {
-            registerNewSending(cardSendings);
-            return true;
+        // Synchronization is needed as :
+        //  - cardSendings is shared between threads
+        //  - cardSendings list is not thread-safe
+        // See https://github.com/opfab/operatorfabric-core/issues/6727
+        synchronized (cardSendings) {
+            if (!isLimitReached(cardSendings)) {
+                registerNewSending(cardSendings);
+                return true;
+            } else
+                return false;
         }
-        else
-            return false;
     }
 
     public void reset() {
@@ -61,7 +65,7 @@ public class CardSendingLimiter {
 
     private void registerNewSending(List<Long> cardSendings) {
         cardSendings.add(customClock.millis());
-        if(cardSendings.size() > limitCardCount)
+        if (cardSendings.size() > limitCardCount)
             cardSendings.remove(0);
     }
 }

